app/connectors/AgentClientMandateFrontendConnector.scala

```diff
@@ -16,54 +16,43 @@
 
 package connectors
 
-import config.WSHttp
+import config.ApplicationConfig
+import javax.inject.Inject
 import models.{AgentEmail, ClientDisplayName, OldMandateReference}
-import play.api.Mode.Mode
 import play.api.mvc.Request
-import play.api.{Configuration, Play}
 import uk.gov.hmrc.crypto.ApplicationCrypto
-import uk.gov.hmrc.http.CoreGet
-import uk.gov.hmrc.play.config.ServicesConfig
-import uk.gov.hmrc.play.frontend.auth.AuthContext
-import uk.gov.hmrc.play.frontend.filters.SessionCookieCryptoFilter
+import uk.gov.hmrc.play.bootstrap.filters.frontend.crypto.CryptoImplicits
+import uk.gov.hmrc.play.bootstrap.http.DefaultHttpClient
 import uk.gov.hmrc.play.partials.HeaderCarrierForPartialsConverter
 
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
-trait AgentClientMandateFrontendConnector extends ServicesConfig with RawResponseReads with HeaderCarrierForPartialsConverter {
+class AgentClientMandateFrontendConnector @Inject()(appConfig: ApplicationConfig,
+                                                    appCrypto: ApplicationCrypto,
+                                                    http: DefaultHttpClient
+                                                   ) extends RawResponseReads with HeaderCarrierForPartialsConverter with CryptoImplicits {
 
-  def serviceUrl = baseUrl("agent-client-mandate-frontend")
+  val serviceUrl: String = appConfig.servicesConfig.baseUrl("agent-client-mandate-frontend")
```

Comment:
seen multiple references to this now, definitely should have a `serviceUrl` entry in ApplicationConfig